### PR TITLE
OSDOCS-348 Registry storage for GCP

### PR DESCRIPTION
**Pull request by bmcelvee**
https://jira.coreos.com/browse/OSDOCS-463

### CODE REVIEW

**Comment by bmcelvee** [comment]
This PR is for GCP IPI

**Comment by vikram-redhat** [comment]
Hey @bmcelvee - this build is failing because it includes a link to the architecture-component-imageregistry.adoc file, which has an error in it, because it is missing an anchor id here: (https://github.com/openshift/openshift-docs/blame/master/registry/architecture-component-imageregistry.adoc#L12).

If you edit that file, add the right anchor to it, and add it as part of this PR, it will fix the build.

File: modules/registry-configuring-storage-gcp-user-infra.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+You must set up the storage medium manually and configure the settings in the
+registry CRD.
+
+.Prerequisites
+
+* To configure registry storage for GCP, you need to provide Registry Operator
+cloud credentials.
+
+.Procedure
+
```
Comment (bmcelvee) [inline modules/registry-configuring-storage-gcp-user-infra.adoc]: Stub for GCP procedure

**Comment by bmcelvee** [comment]
@coreydaley - I'm working on updating the registry storage docs as we discussed on Slack and moved the "Image Registry Operator configuration resources" module to preface the modules for AWS and GCP storage. Right now it's a bit sparse and I've just stubbed in the procedure section for GCP (it's commented out currently). 

But I figured this would be a good start. I looked through the code PR, but didn't immediately see information for starting the procedure. Would you be able (and willing :smile_cat:) to point me to where I might find that information?
 
https://osdocs-463-registry-storage-gcp--ocpdocs.netlify.com/openshift-enterprise/latest/registry/configuring-registry-operator.html#configuring-registry-storage-aws

Thanks!

File: registry/configuring-registry-operator.adoc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[id=configuring-registry-operator’]
+[id="configuring-registry-operator"]
 = Image Registry Operator in {product-title}
 include::modules/common-attributes.adoc[]
 :context: configuring-registry-operator
```
Comment (bmcelvee) [inline registry/configuring-registry-operator.adoc]: I think the following note needs to be updated: 

Storage is only automatically configured when you install on Amazon Web Services.

Comment (coreydaley) [reply registry/configuring-registry-operator.adoc]: Storage for the registry is automatically configured when doing an IPI install on AWS, GCP, or Azure.

File: modules/registry-operator-config-resources-secret.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+
+[id="registry-operator-config-resources-secret_{context}"]
+= Configuring a secret for the Image Registry Operator
+
+In addition to the `configs.imageregistry.operator.openshift.io` and ConfigMap
+resources, configuration is provided to the Operator by a separate Secret
+resource located within the `openshift-image-registry` namespace.
+
+The `image-registry-private-configuration` Secret provides
```
Comment (coreydaley) [inline modules/registry-operator-config-resources-secret.adoc]: image-registry-private-configuration-user

**Comment by bmcelvee** [comment]
@openshift/team-documentation PTAL

File: modules/registry-configuring-storage-aws-user-infra.adoc [hunk 4]
```diff
@@ -16,6 +17,9 @@ procedure.
 .Prerequisites
 
 * A cluster on AWS with user-provisioned infrastructure.
+* For S3 on AWS storage the ConfigMap is expected to contain two keys:
```
Comment (coreydaley) [inline modules/registry-configuring-storage-aws-user-infra.adoc]: This should be a Secret, not a ConfigMap

File: modules/registry-operator-configuration-resource-overview-aws-s3.adoc [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+The following configuration parameters are available for AWS S3 registry storage.
+
+[cols="3a,8a",options="header"]
+|===
+|Parameter |Description
+
+|`ImageRegistryConfigStorageS3`
+|ImageRegistryConfigStorageS3 holds the information to configure the registry
+to use the AWS S3 service for backend storage.
```
Comment (coreydaley) [inline modules/registry-operator-configuration-resource-overview-aws-s3.adoc]: I think that the above three lines should be omitted.

File: modules/registry-operator-configuration-resource-overview-gcp-gcs.adoc [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+The following configuration parameters are available for GCP GCS registry storage.
+
+[cols="3a,8a",options="header"]
+|===
+|Parameter |Description
+
+|`ImageRegistryConfigStorageGCS`
+|ImageRegistryConfigStorageGCS holds the information to configure the registry
+to use the GCP GCS service for backend storage.
```
Comment (coreydaley) [inline modules/registry-operator-configuration-resource-overview-gcp-gcs.adoc]: I think that the above three lines should be omitted.

File: modules/registry-configuring-storage-aws-user-infra.adoc [hunk 7]
```diff
@@ -2,6 +2,7 @@
 //
 //* architecture/installation-.adoc
 // * installing/installing_aws_user_infra/installing-aws-user-infra.adoc
+// * registry/configuring-registry-storage-aws-user-infrastructure.adoc
 
 [id="registry-configuring-storage-aws-user-infra_{context}"]
 = Configuring registry storage for AWS with user-provisioned infrastructure
```
Comment (xiuwang) [inline modules/registry-configuring-storage-aws-user-infra.adoc]: s/an S3 bucket/a S3 bucket  i
in the following field.

File: modules/registry-operator-config-resources-secret.adoc [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+
+[id="registry-operator-config-resources-secret_{context}"]
+= Configuring a Secret for the Image Registry Operator
+
+In addition to the `configs.imageregistry.operator.openshift.io` and ConfigMap
+resources, configuration is provided to the Operator by a separate Secret
+resource located within the `openshift-image-registry` namespace.
+
+The `image-registry-private-configuration-user` Secret provides
```
Comment (xiuwang) [inline modules/registry-operator-config-resources-secret.adoc]: Now, the secret named 'image-registry-private-configuration'

**Review by sheriff-rh — APPROVED** [review]
Looks great!

File: modules/registry-configuring-storage-aws-user-infra.adoc (hunk 7 above)
Comment (bmcelvee) [reply modules/registry-configuring-storage-aws-user-infra.adoc]: @xiuwang Where should the s/in the following field go? I'm sorry to ask, but for some reason the comment is showing up in a different place in my Git view.

**Comment by bmcelvee** [comment]
Thank you, all, for the reviews! I'm going ahead and merging this PR so we can continue to add registry storage content. We can also continue iterating on the GCP content in a follow-up PR.

File: modules/registry-operator-config-resources-configmap.adoc [hunk 9]
```diff
@@ -68,3 +88,23 @@ sh-4.2
...
+    -----BEGIN CERTIFICATE-----
+    ...
+    -----END CERTIFICATE-----
+  registry-with-port.example.com..5000: | <1>
+    -----BEGIN CERTIFICATE-----
+    ...
+    -----END CERTIFICATE-----
+----
+<1>  If the registry has the port, such as `registry-with-port.example.com:5000`,
+`:` should be replaced with `...`.
```
Comment (xiuwang) [inline modules/registry-operator-config-resources-configmap.adoc]: `...` should be two dot `..`

File: modules/registry-configuring-storage-aws-user-infra.adoc (hunk 7 above)
Comment (xiuwang) [reply modules/registry-configuring-storage-aws-user-infra.adoc]: This issue is resolved in https://github.com/openshift/openshift-docs/pull/16593/files#diff-f981c0ddf4f00dc6b26823c9f04d898cL14

File: modules/registry-operator-configuration-resource-overview-gcp-gcs.adoc [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+|Bucket is the bucket name in which you want to store the registry's data.
+It is optional and is generated if not provided.
+
+|`region`
+|Region is the GCS location in which your bucket exists. It is optional and is
+set based on the installed GCS Region.
+
+|`regionEndpoint`
+|RegionEndpoint is the endpoint for S3 compatible storage services.
+It is optional and defaults based on the Region that is provided.
```
Comment (xiuwang) [inline modules/registry-operator-configuration-resource-overview-gcp-gcs.adoc]: Should remove regionEndpoint  from doc, the paramter is using for S3, but GCS

**Pull request merged**